app/components/location_subject_filter_component.rb

```diff
@@ -0,0 +1,13 @@
+class LocationSubjectFilterComponent < ViewComponent::Base
+  include ViewHelper
+
+  attr_reader :results
+
+  def initialize(results:)
+    @results = results
+  end
+
+  def subjects_line
+    @results.subjects.map { |subject| "<b>#{subject[:subject_name]}</b>" }.to_sentence(last_word_connector: ' and ').html_safe
```

Comment:
Using Rails’s version of `to_sentence` and the `tag` helper might avoid the need to mark the string as `html_safe`?

```suggestion
    to_sentence(@results.subjects.map { |subject| tag.b(subject[:subject_name) }, last_word_connector: ' and ')
```